src/WebJobs.Extensions.DurableTask/Listener/TaskOrchestrationShim.cs

```diff
@@ -117,7 +124,60 @@ public void SetFunctionInvocationCallback(Func<Task> callback)
 
             if (returnValue != null)
             {
-                this.context.SetOutput(returnValue);
+                if (orchestratorInfo.IsOutOfProc && returnValue is JObject)
+                {
+                    var jObj = returnValue as JObject;
+                    if (jObj["actions"] != null)
+                    {
+                        var commandSets = JsonConvert.DeserializeObject<AsyncCommand[][]>(jObj["actions"].ToString());
+
+                        foreach (AsyncCommand[] commandSet in commandSets)
```

Comment:
I suggest we add some comments describing what the array of arrays represents - e.g. the first dimension represents a particular episode of the orchestration and the second dimension represents all the actions that were scheduled within that episode (I hope I'm understanding it correctly).